extract-wos-article.py

```diff
@@ -23,7 +30,54 @@
     "Current Publisher": "publisher",
 }
 
-YEAR_REGEX = re.compile(".*([0-9]{4})")
+REGEX = {
+    "year": re.compile(".*([0-9]{4})"),
+    "wos_id_number": re.compile("([0-9]{15})"),
+}
+CITING_SUMMARY_URL_PREFIX = "https://www.webofscience.com/wos/woscc/citing-summary/"
+
+
+async def extract_citing_summary(page: Page, article_id: str):
+    url = CITING_SUMMARY_URL_PREFIX + article_id
+    await page.goto(url, wait_until="networkidle")
+
+    curr_page = await page.query_selector(CITING_SELECTORS["current_page"])
+    curr_page_text = await curr_page.input_value()
+    curr_page_num = int(curr_page_text.strip())
+
+    final_page = await page.query_selector(CITING_SELECTORS["final_page"])
+    final_page_text = await final_page.inner_html()
+    final_page_num = int(final_page_text.strip())
+
+    articles = []
+    while True:
+        curr_articles_num = -1
+        curr_articles = []
+
+        while len(curr_articles) > curr_articles_num:
+            curr_articles_num = len(curr_articles)
+            await page.keyboard.press("PageDown")
+            curr_articles = await page.query_selector_all(
+                CITING_SELECTORS["citations_articles"]
+            )
+            time.sleep(0.1)
+
+        articles.extend(curr_articles)
+
+        if curr_page_num < final_page_num:
+            await page.click('[aria-label="Bottom Next Page"]')
+            time.sleep(2)
+        if curr_page_num == final_page_num:
+            break
+
+        curr_page_num += 1
+
+    articles_ids = []
+    for article in articles:
+        article_text = await article.inner_html()
+        article_id = REGEX["wos_id_number"].findall(article_text)[0]
```

Comment:
I am a bit afraid here that we might be able to find a 15 `[0-9]` numbers in a couple of places, many of which most probably won't be WOS IDs. Is there a chance we could base the detection on something other than just the list of numbers here? (Sorry, I do not really know what would the `article_text` normally hold)

Reply:
@mrshu `article_text` always contains only `<a>` element as below:
```html
<!---->
<!---->
<a data-ta="summary-record-title-link" class="title title-link font-size-18 ng-star-inserted" 
href="/wos/woscc/full-record/WOS:000718302600032">
Rapid screening for SARS-CoV-2 VOC-Alpha (202012/01, B.1.1.7) using the Allplex (TM) SARS-CoV-2/FluA/FluB/RSV Assay
</a>
<!---->
<!---->
<!---->
```

I've adjusted the regex from `([0-9]{15})` to `(WOS:[0-9]{15})`, so it does not extract some random numbers. still, if there are somehow more of the WOS IDs it would be problem, but for now, it think it serves well enough. i can look at it more deeply later

Reply:
Cool, this works rather nicely :slightly_smiling_face: